layers/+lang/markdown/packages.el

```diff
@@ -65,6 +65,18 @@
       ;; Make markdown-mode behave a bit more like org w.r.t. code blocks i.e.
       ;; use proper syntax highlighting
       (setq markdown-fontify-code-blocks-natively t)
+      ;; set the markdown-command
+      (if markdown-executable     ; verify the executable for user defined value
+          (if (executable-find markdown-executable)
+              (setq markdown-command markdown-executable)
+            (warn "Non-exist markdown-executable: %s" markdown-executable))
+        (when (or (null markdown-command) ; markdown-command is nil or not exist
+                  (not (executable-find markdown-command)))
+          ;; try more markdown CLIs
+          (let ((cmd (cl-loop for cmd in '("markdown_py")
+                              when (executable-find cmd)
+                              return (file-name-nondirectory it))))
+            (when cmd (setq markdown-command cmd)))))
```

Comment:
I think we can dramatically reduce the complexity of this block by using the suggested code block from lebensterben with some slight changes:

```suggestion
      ;; set the markdown-command
(setq markdown-command
      (if-let ((command (cl-loop for cmd in (if markdown-preview-command
                                                (list markdown-preview-command)
                                              (list "markdown" "pandoc" "markdown_py"))
                                 when (executable-find cmd)
                                 return (file-name-nondirectory it))))
          command
        (user-error (if markdown-preview-command
                        (format "Cannot find markdown-preview-command %s" markdown-preview-command)
                      "Please refer to the layer documentation and install a markdown command"))))
```
This block will only check the entered `markdown-executable` if maintained but still fallback to one of the defaults if this is not so. Also it throws a `user-error` which I very much prefer to a subtle warning in this case.

Last but not least its following a functional approach which, at least in the lisp languages, is preferred over an imperative one both in performance and maintainability.

Reply:
It's okay for me. According the definition of `markdown-command`, it will try two commands `markdown` and `pandoc`, this command list maybe growth to 3 or more commands in the future. So in my code, will respect the result of `markdown-command` initialize, when it's not exist, try list `(markdown_py)` as last choice.